vividus-plugin-web-app/src/test/java/org/vividus/spring/StringToIntegerRangeConverterTests.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2019-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.spring;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.vividus.model.IntegerRange;
+
+class StringToIntegerRangeConverterTests
+{
+    private final StringToIntegerRangeConverter converter = new StringToIntegerRangeConverter();
+
+    static Stream<Arguments> rangeSource()
+    {
+        return Stream.of(
+                arguments("0..5", List.of(0, 1, 2, 3, 4, 5)),
+                arguments("4,5", List.of(4, 5)),
+                arguments("-1..4,3..6", List.of(-1, 0, 1, 2, 3, 4, 5, 6)),
+                arguments("3..6,-1..4", List.of(3, 4, 5, 6, -1, 0, 1, 2))
```

Comment:
the valid port range is 0-65535, shouldn't `IntegerRange` be renamed to `PortsRange`? (also additional validations should be added)

Reply:
The IntegerRange itself I wanted to make generic, as to port verification that's verified in underlying layer by java.net.InetSocketAddress.checkPort(int)

Reply:
agree, it's better to check ports before they are pulled from the ports pool